File: python/151-160/Intersection of Two Linked Lists.py
    # @param two ListNodes
    # @return the intersected ListNode
    def getIntersectionNode(self, headA, headB):
        if headA==None or headB==None:
            return None
        pa,pb=headA,headB
        sa,sb=1,1
        while pa.next!=None:
            sa+=1
            pa=pa.next
        while pb.next!=None:
            sb+=1
            pb=pb.next
        if pa!=pb:
            return None
        pa,pb=headA,headB
        if sa>sb:
            t=sa-sb
            while t>0:
                pa=pa.next
                t-=1
        else:
            t=sb-sa
            while t>0:
                pb=pb.next
                t-=1
        while pa!=pb:
            pa=pa.next
            pb=pb.next
        return pa

File: Cpp/151-160/Intersection of Two Linked Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
 /*
    runtime: 52ms
    errot:0
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        if(headA==NULLL || headB==NULL) return NULL;
        int lenA = 1, lenB=1; 
        ListNode* tA = headA,*tB = headB;
        while(tA->next!=NULL){
            lenA++; tA=tA->next;
        }
        while(tB->next!=NULL){
            lenB++; tB=tB->next;
        }
        if(tA!=tB) return NULL;
        int i=0; 
        if(lenA>lenB){
            while(i<lenA-lenB)
            { headA=headA->next; i++;}
        }else{
            while(i<lenB-lenA)
            { headB=headB->next; i++; }
        }
        while(headA!=headB){
            headA=headA->next;
            headB=headB->next;
        }
        return headA;
    }
};
